app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+import fetch from 'isomorphic-fetch';
+import React, {Component} from 'react';
+import {render} from 'react-dom';
+import {baseUrl, isWebBrowser, dynamicRequireElectron} from '../utils/utils';
+
+const CLOUD = 'cloud';
+const ONPREM = 'onprem';
+const SERVER_TYPES = {
+    [CLOUD]: 'Plotly Cloud',
+    [ONPREM]: 'Plotly On-Premise'
+};
+
+class Setup extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            domain: '',
+            errorMessage: '',
+            loggedIn: false,
+            serverType: '',
+            status: '',
+            username: ''
+        };
+        this.authenticateUser = this.authenticateUser.bind(this);
+        this.buildOauthUrl = this.buildOauthUrl.bind(this);
+        this.chooseServerType = this.chooseServerType.bind(this);
+        this.setTimeoutAuthDone = this.setTimeoutAuthDone.bind(this);
+        this.updateStateWithEvent = this.updateStateWithEvent.bind(this);
+        this.verifyAuthDone = this.verifyAuthDone.bind(this);
+    }
+
+    buildOauthUrl() {
+        const oauthClientId = 'isFcew9naom2f1khSiMeAtzuOvHXHuLwhPsM7oPt';
+        const isOnPrem = this.state.serverType === ONPREM;
+        const plotlyDomain = isOnPrem ? this.state.domain : 'plot.ly';
+        const redirect_uri = baseUrl().replace('/setup', '');
+        return (
+            `https://${plotlyDomain}/o/authorize/?response_type=token&` +
```

Comment:
the user may enter `https://plotly.acme.com` if they just copy and paste the URL - should trip this for them.

Also, what happens if the user isn't logged in to their plotly account, do we need to redirect them through the `/accounts/login?redirect_url='/o/authorize` or does the plotly server properly handle logging them in?